taskvine/src/manager/vine_manager.c

```diff
@@ -1461,9 +1461,10 @@ static vine_msg_code_t handle_taskvine(struct vine_manager *q, struct vine_worke
 {
 	char items[4][VINE_LINE_MAX];
 	int worker_protocol;
+	int worker_library_protocol;
 
-	int n = sscanf(line, "taskvine %d %s %s %s %s", &worker_protocol, items[0], items[1], items[2], items[3]);
-	if (n != 5)
+	int n = sscanf(line, "taskvine %d %d %s %s %s %s", &worker_protocol, &worker_library_protocol, items[0], items[1], items[2], items[3]);
```

Comment:
I'm not sure about having a different number for the library protocol.  I'd keep only the worker_protocol and increment it if the library communication between the worker and library changes.